extensions/amp-soundcloud/0.1/amp-soundcloud.js

```diff
@@ -0,0 +1,81 @@
+/**
+ * Copyright 2016 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+/**
+ * @fileoverview Embeds a Soundcloud clip
+ *
+ * Example:
+ * <code>
+ * <amp-soundcloud
+ *   height=166
+ *   data-trackid="243169232"
+ *   data-color="ff5500"
+ *   data-autoplay="true"
+ *   layout="fixed-height">
+ * </amp-soundcloud>
+ *
+ *
+ */
+
+import {isLayoutSizeDefined} from '../../../src/layout';
+import {loadPromise} from '../../../src/event-helper';
+
+
+class AmpSoundcloud extends AMP.BaseElement {
+  /** @override */
+  preconnectCallback(onLayout) {
+    this.preconnect.url('https://api.soundcloud.com/', onLayout);
+  }
+
+  /** @override */
+  isLayoutSupported(layout) {
+    return isLayoutSizeDefined(layout);
```

Comment:
`fixed-height` along is a bit unusual, so please double-check. It's for things like scrolling carousel that need to grow horizontally, but not vertically.